src/EnergyPlus/ChillerElectricASHRAE205.cc

```diff
@@ -0,0 +1,1665 @@
+    } else {
+        if (message_type == tk205::MsgSeverity::WARN_205) {
+            ShowWarningError(*contextPair.first, fullMessage);
+        } else if (message_type == tk205::MsgSeverity::INFO_205) {
+            ShowMessage(*contextPair.first, fullMessage);
+        } else {
+            ShowMessage(*contextPair.first, fullMessage);
+        }
+    }
+}
+
+void getChillerASHRAE205Input(EnergyPlusData &state)
+{
+    static constexpr std::string_view RoutineName("getChillerASHRAE205Input: "); // include trailing blank space
+
+    using namespace tk205;
+    RSInstanceFactory::register_factory("RS0001", std::make_shared<RS0001Factory>());
+
+    bool ErrorsFound{false};
+
+    state.dataIPShortCut->cCurrentModuleObject = ChillerElectricASHRAE205::ASHRAE205ChillerSpecs::ObjectType;
+    auto &ip = state.dataInputProcessing->inputProcessor;
+    int numElectric205Chillers = ip->getNumObjectsFound(state, state.dataIPShortCut->cCurrentModuleObject);
+
+    if (numElectric205Chillers <= 0) {
+        ShowSevereError(state, format("No {} equipment specified in input file", state.dataIPShortCut->cCurrentModuleObject));
+        ErrorsFound = true;
+    }
+
+    state.dataChillerElectricASHRAE205->Electric205Chiller.allocate(numElectric205Chillers);
+
+    auto const ChillerInstances = ip->epJSON.find(state.dataIPShortCut->cCurrentModuleObject).value();
+    auto ChillerNum{0};
+    auto const &objectSchemaProps = ip->getObjectSchemaProps(state, state.dataIPShortCut->cCurrentModuleObject);
+    for (auto &instance : ChillerInstances.items()) {
+        auto const &fields = instance.value();
+        auto const &thisObjectName = instance.key();
+        GlobalNames::VerifyUniqueChillerName(
+            state, state.dataIPShortCut->cCurrentModuleObject, thisObjectName, ErrorsFound, state.dataIPShortCut->cCurrentModuleObject + " Name");
+
+        ++ChillerNum;
+        auto &thisChiller = state.dataChillerElectricASHRAE205->Electric205Chiller(ChillerNum);
+        thisChiller.Name = UtilityRoutines::MakeUPPERCase(thisObjectName);
+        ip->markObjectAsUsed(state.dataIPShortCut->cCurrentModuleObject, thisObjectName);
+
+        auto rep_file_name = ip->getAlphaFieldValue(fields, objectSchemaProps, "representation_file_name");
+        fs::path rep_file_path = DataSystemVariables::CheckForActualFilePath(state, fs::path(rep_file_name), std::string(RoutineName));
+        std::pair<EnergyPlusData *, std::string> callbackPair{&state,
+                                                              format("{} \"{}\"", state.dataIPShortCut->cCurrentModuleObject, thisObjectName)};
+        tk205::set_error_handler(tk205ErrCallback, &callbackPair);
+        thisChiller.Representation =
+            std::dynamic_pointer_cast<tk205::rs0001_ns::RS0001>(RSInstanceFactory::create("RS0001", rep_file_path.string().c_str()));
+        if (nullptr == thisChiller.Representation) {
```

Comment:
`if (!thisChiller.Representation)`?  Not a big deal, but it feels a little like `if (false == someFlag)`.

Reply:
If `if (!stuff)` is preferred, I'll change this - but since C++11 introduced `nullptr_t` as its own type, I tend to use it explicitly rather than relying on implicit "falsy" conversions.
Also happy to move the order of comparison around if this is not the E+ way. (I waffle on which I like better.)

Reply:
I would say cases like this, the "not null" check, are one type of truthy checks that I actually prefer over being explicit and comparing it to null.  But it's nothing to hold up the PR.  Moving on.